diff --git a/ollamarama.py b/ollamarama.py
index 6aeef13..cd2d0f1 100644
--- a/ollamarama.py
+++ b/ollamarama.py
@@ -11,16 +11,24 @@ import datetime
 import asyncio
 import requests
 import markdown
+import sys
 
 class ollamarama:
-    def __init__(self):
+    def __init__(self, config_file):
         #load config file
-        self.config_file = "config.json"
+        self.config_file = config_file
         with open(self.config_file, "r") as f:
             config = json.load(f)
             f.close()
 
         self.server, self.username, self.password, self.channels, self.admins = config["matrix"].values()
+        self.name = self.username.split(':')[0][1:]
+
+        #load character file
+        self.character_file = f"/var/lib/ollamarama/characters/{self.name}.json"
+        with open(self.character_file, "r") as f:
+            character = json.load(f)
+            f.close()
 
         self.client = AsyncClient(self.server, self.username)
 
@@ -28,7 +36,14 @@ class ollamarama:
         self.join_time = datetime.datetime.now()
         
         # store chat history
-        self.messages = {}
+        self.history_file = f"/var/lib/ollamarama/history/{self.name}.json"
+
+        try:
+            with open(self.history_file, "r") as f:
+                self.messages = json.load(f)
+                f.close()
+        except IOError:
+            self.messages = {}
 
         self.api_url = config["ollama"]["api_base"] + "/api/chat"
 
@@ -43,9 +58,9 @@ class ollamarama:
             "repeat_penalty": self.repeat_penalty
         }
 
-        self.default_personality = config["ollama"]["personality"]
+        self.default_personality = character["personality"]
         self.personality = self.default_personality
-        self.prompt = config["ollama"]["prompt"]
+        self.prompt = character["prompt"]
 
     # get the display name for a user
     async def display_name(self, user):
@@ -84,6 +99,10 @@ class ollamarama:
             else:
                 del self.messages[channel][sender][0:2]
 
+        with open(self.history_file, "w") as f:
+            json.dump(self.messages, f)
+            f.close()
+
     #generate Ollama model response
     async def respond(self, channel, sender, message, sender2=None):
         try:
@@ -111,11 +130,10 @@ class ollamarama:
             # .x function was used
             if sender2:
                 display_name = await self.display_name(sender2)
+                response_text = f"**{display_name}**:\n{response_text.strip()}"
             # .ai was used
             else:
-                display_name = await self.display_name(sender)
-
-            response_text = f"**{display_name}**:\n{response_text.strip()}"
+                response_text = f"{response_text.strip()}"
             
             try:
                 await self.send_message(channel, response_text)
@@ -205,6 +223,10 @@ class ollamarama:
         self.personality = self.default_personality
         self.temperature, self.top_p, self.repeat_penalty = self.defaults.values()
         await self.send_message(channel, "Bot has been reset for everyone")
+
+        with open(self.history_file, "w") as f:
+            json.dump(self.messages, f)
+            f.close()
         
     async def handle_message(self, message, sender, sender_display, channel):
         user_commands = {
@@ -227,9 +249,12 @@ class ollamarama:
         if command in user_commands:
             action = user_commands[command]
             await action()
-        if sender_display in self.admins and command in admin_commands:
+        elif sender_display in self.admins and command in admin_commands:
             action = admin_commands[command]
             await action()
+        else:
+            action = user_commands[".ai"]
+            await action()
         
     async def message_callback(self, room: MatrixRoom, event: RoomMessageText):
         if isinstance(event, RoomMessageText):
@@ -259,7 +284,7 @@ class ollamarama:
         for channel in self.channels:
             try:
                 await self.client.join(channel)
-                print(f"{self.bot_id} joined {channel}")
+                print(f"{self.username} joined {channel}")
                 
             except:
                 print(f"Couldn't join {channel}")
@@ -270,6 +295,6 @@ class ollamarama:
         await self.client.sync_forever(timeout=30000, full_state=True) 
 
 if __name__ == "__main__":
-    ollamarama = ollamarama()
-    asyncio.get_event_loop().run_until_complete(ollamarama.main())
+    ollamarama = ollamarama(sys.argv[1])
+    asyncio.run(ollamarama.main())
